DQMOffline/Trigger/plugins/JetMETMonitor.cc

```diff
@@ -0,0 +1,1056 @@
+  JetMET_N_PHI, -JetMET_MAX_PHI, JetMET_MAX_PHI
+};
+// Define Eta Bin //
+double JetMET_MAX_ETA = 5;
+int JetMET_N_ETA = 50;
+MEbinning jetmet_eta_binning_{
+  JetMET_N_ETA, -JetMET_MAX_ETA, JetMET_MAX_ETA
+};
+
+
+
+// -----------------------------
+//  constructors and destructor
+// -----------------------------
+
+JetMETMonitor::JetMETMonitor( const edm::ParameterSet& iConfig ):
+   metSelection_ ( iConfig.getParameter<std::string>("metSelection") )
+  , jetSelection_ ( iConfig.getParameter<std::string>("jetSelection") )
+  , calojetSelection_ ( iConfig.getParameter<std::string>("calojetSelection") )
+  , eleSelection_ ( iConfig.getParameter<std::string>("eleSelection") )
+  , muoSelection_ ( iConfig.getParameter<std::string>("muoSelection") ) 
+{
+  folderName_            = iConfig.getParameter<std::string>("FolderName"); 
+  metToken_              = consumes<reco::PFMETCollection>        (iConfig.getParameter<edm::InputTag>("met")       );
+  pfjetToken_            = mayConsume<reco::PFJetCollection>      (iConfig.getParameter<edm::InputTag>("pfjets")    ); 
+  calojetToken_          = mayConsume<reco::CaloJetCollection>    (iConfig.getParameter<edm::InputTag>("calojets")    ); 
+  eleToken_              = mayConsume<reco::GsfElectronCollection>(iConfig.getParameter<edm::InputTag>("electrons") );
+  muoToken_              = mayConsume<reco::MuonCollection>       (iConfig.getParameter<edm::InputTag>("muons")     );     
+  met_variable_binning_  = iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<std::vector<double> >("jetptBinning");
+  met_binning_           = getHistoPSet   (iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<edm::ParameterSet>   ("metPSet")    );
+  ls_binning_            = getHistoLSPSet (iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<edm::ParameterSet>   ("lsPSet")     );
+
+  num_genTriggerEventFlag_ = new GenericTriggerEventFlag(iConfig.getParameter<edm::ParameterSet>("numGenericTriggerEventPSet"),consumesCollector(), *this);
+  den_genTriggerEventFlag_ = new GenericTriggerEventFlag(iConfig.getParameter<edm::ParameterSet>("denGenericTriggerEventPSet"),consumesCollector(), *this);
+
+  njets_      = iConfig.getParameter<int>("njets" );
+  nelectrons_ = iConfig.getParameter<int>("nelectrons" );
+  nmuons_     = iConfig.getParameter<int>("nmuons" );
+  ptcut_      = iConfig.getParameter<double>("ptcut" ); // for HLT Jet 
+  isPFJetTrig    = iConfig.getParameter<bool>("ispfjettrg" );
+  isCaloJetTrig  = iConfig.getParameter<bool>("iscalojettrg" );
+  isMetTrig      = iConfig.getParameter<bool>("ismettrg" );
+  jetmetME_.numerator   = nullptr;
+  jetmetME_.denominator = nullptr;
+  jetmetME_variableBinning_.numerator   = nullptr;
+  jetmetME_variableBinning_.denominator = nullptr;
+  jetmetVsLS_.numerator   = nullptr;
+  jetmetVsLS_.denominator = nullptr;
+  jetmetPhiME_.numerator   = nullptr;
+  jetmetPhiME_.denominator = nullptr;
+  if (isMetTrig == false) 
```

Comment:
I thought I want to make a code to cover jet trigger and met trigger. 
But It's not efficient way. Your suggestion is better than my code.
So I'm trying to modify this code for only jet. 
And for met trigger, I will use METMonitor code. 
I need to change this code into JetMonitor.cc . Do you agree with me?